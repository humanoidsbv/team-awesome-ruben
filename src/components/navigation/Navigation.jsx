import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';

import styles from './Navigation.module.css';

const Navigation = ({ isMenuVisible }) => (
  <nav className={`${styles.nav} ${isMenuVisible ? styles.navActive : ''}`}>
    <Link to="/" className={styles.item} activeClassName={styles.itemActive}>
      Time entries
    </Link>
    <Link
      activeClassName={styles.itemActive}
      className={styles.item}
      to="/team-members"
    >
      Team members
    </Link>
    <button type="button" className={styles.item}>
      Projects
    </button>
    <Link
      activeClassName={styles.itemActive}
      className={styles.item}
      to="/clients"
    >
      Clients
    </Link>
    <button type="button" className={styles.item} href="/">
      Documents
    </button>
  </nav>
);

Navigation.propTypes = {
  isMenuVisible: PropTypes.bool.isRequired
};

export default Navigation;
