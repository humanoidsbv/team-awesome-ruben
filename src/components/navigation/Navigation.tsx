import React from 'react';
import { Link } from 'gatsby';

import styles from './Navigation.module.css';

interface NavigationProps {
  isMenuVisible: boolean;
}

const Navigation = ({ isMenuVisible }: NavigationProps): React.ReactElement => (
  <nav className={`${styles.nav} ${isMenuVisible ? styles.navActive : ''}`}>
    <Link to="/" className={styles.item} activeClassName={styles.itemActive}>
      Time entries
    </Link>
    <Link
      activeClassName={styles.itemActive}
      className={styles.item}
      to="/team-members"
    >
      Team members
    </Link>
    <button type="button" className={styles.item}>
      Projects
    </button>
    <Link
      activeClassName={styles.itemActive}
      className={styles.item}
      to="/clients"
    >
      Clients
    </Link>
    <button type="button" className={styles.item}>
      Documents
    </button>
  </nav>
);

export default Navigation;
