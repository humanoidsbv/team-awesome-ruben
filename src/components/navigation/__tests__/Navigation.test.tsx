import * as React from 'react';
import { shallow } from 'enzyme';

import Navigation from '../Navigation';

it('matches component output with expected output', () => {
  const navigation = shallow(<Navigation isMenuVisible={false} />);
  expect(navigation).toMatchSnapshot();
});
