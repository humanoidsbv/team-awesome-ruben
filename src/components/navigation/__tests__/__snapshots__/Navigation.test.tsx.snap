// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component output with expected output 1`] = `
<nav
  className="nav "
>
  <ForwardRef
    activeClassName="itemActive"
    className="item"
    to="/"
  >
    Time entries
  </ForwardRef>
  <ForwardRef
    activeClassName="itemActive"
    className="item"
    to="/team-members"
  >
    Team members
  </ForwardRef>
  <button
    className="item"
    type="button"
  >
    Projects
  </button>
  <ForwardRef
    activeClassName="itemActive"
    className="item"
    to="/clients"
  >
    Clients
  </ForwardRef>
  <button
    className="item"
    type="button"
  >
    Documents
  </button>
</nav>
`;
